### DISCUSSION TITLE
Is it necessary to support excessively short keepalive?

### DISCUSSION THREAD

**Discussion by werbenhu** [mochi-mqtt/server#294] [Ideas]
As seen in #293 , users sometimes set a very short keepalive (e.g., keepalive=2) without understanding its purpose. In situations where the network experiences fluctuations, it's easy for timeouts to occur, leading to disconnections. Is it necessary to implement special handling for connections with excessively short keepalive? 

Can we implement some measures, such as actively disconnecting the client by the server if keepalive is set to < 5?

**Comment by dgduncan** [comment]
I have noticed a few people asking questions about the keepalive and not really understanding its' purpose recently. I can't say I am particularly in favor of disconnecting a client based on their keepalive setting as any non-negative value is in spec as a valid keepalive. However, adding an `warning` level log that is present when keepalive < 5 that says "A keepalive value < 5 can cause instability please use with caution" or something like that may be helpful.

**Comment by thedevop** [comment]
We can add MinimumKeepalive in Capabilities. This will be used when client did set a keepalive value and is less than the MinimumKeepalive value.
